### PR TITLE
chore: Cache swift packages - WPB-9913

### PR DESCRIPTION
**Pull request by KaterinaWire**
### Issue
As we adding more and more swift packages the CI should cache (them similar to Carthage) instead of downloading all packages every time.

### Solution

Add steps to the workflow (similar to Carthage) for caching and Bootstrap Swift bundles:
- `Restore Swift Package Dependencies Cache`
- `Resolve Swift Package Dependencies`

And to pass this cache to the` build_app(...) `.

As a cleaning:
- remove ../WireSystem.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved

### Testing

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: .github/workflows/_reusable_app_release.yml [hunk 1]
```diff
@@ -154,10 +156,59 @@ jobs:
       - name: Bootstrap Carthage if no cache
         if: steps.cache-carthage.outputs.cache-hit != 'true'
         run: ./scripts/carthage.sh bootstrap --platform ios --use-xcframeworks
+        # Cache for the first folder (scripts)
+      - name: Restore Swift Package Dependencies (scripts)
+        uses: actions/cache@v3
+        id: cache-scripts-package
+        with:
+          path: scripts/.build
+          key: ${{ runner.os }}-swiftpm-${{ hashFiles('scripts/Package.resolved') }}
```
Comment (netbe) [inline .github/workflows/_reusable_app_release.yml]:
```suggestion
          key: ${{ runner.os }}-swiftpm-scripts-${{ hashFiles('scripts/Package.resolved') }}
```

File: .github/workflows/_reusable_app_release.yml [hunk 2]
```diff
@@ -154,10 +156,59 @@ jobs:
...
+
+        # Cache for the second folder (SourceryPlugin)
+      - name: Restore Swift Package Dependencies (SourceryPlugin)
+        uses: actions/cache@v3
+        id: cache-sourceryPlugin-package
+        with:
+          path: SourceryPlugin/.build
+          key: ${{ runner.os }}-swiftpm-${{ hashFiles('SourceryPlugin/Package.resolved') }}
```
Comment (netbe) [inline .github/workflows/_reusable_app_release.yml]:
```suggestion
          key: ${{ runner.os }}-swiftpm-sourcery-${{ hashFiles('SourceryPlugin/Package.resolved') }}
```

File: .github/workflows/_reusable_app_release.yml [hunk 3]
```diff
@@ -154,10 +156,59 @@ jobs:
...
           bundler-cache: true
+          #dowland dep
+          #r
+
+        # Restore Swift Package Dependencies Cache
+      - name: Restore Swift Package Dependencies Cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.PACKAGES_DIR }}
+          key: ${{ runner.os }}-swiftpm-${{ hashFiles('**/Package.resolved') }}
```
Comment (netbe) [inline .github/workflows/_reusable_app_release.yml]:
```suggestion
          key: ${{ runner.os }}-swiftpm-project-${{ hashFiles('**/Package.resolved') }}
```

File: .github/workflows/_reusable_app_release.yml [hunk 4]
```diff
@@ -154,6 +156,46 @@ jobs:
...
         if: steps.cache-carthage.outputs.cache-hit != 'true'
         run: ./scripts/carthage.sh bootstrap --platform ios --use-xcframeworks
+
+        # Cache for the first folder (scripts)
+      - name: Restore Swift Package Dependencies (scripts)
+        uses: actions/cache@v3
+        id: cache-scripts-package
+        with:
+          path: scripts/.build
+          key: ${{ runner.os }}-swiftpm-scripts-${{ hashFiles('scripts/Package.resolved') }}
```
Comment (KaterinaWire) [inline .github/workflows/_reusable_app_release.yml:166]: the key here is `macOS-swiftpm-scripts-`

File: .github/workflows/_reusable_app_release.yml [hunk 5]
```diff
@@ -154,6 +156,46 @@ jobs:
...
+
+        # Cache for the second folder (SourceryPlugin)
+      - name: Restore Swift Package Dependencies (SourceryPlugin)
+        uses: actions/cache@v3
+        id: cache-sourceryPlugin-package
+        with:
+          path: SourceryPlugin/.build
+          key: ${{ runner.os }}-swiftpm-sourcery-${{ hashFiles('SourceryPlugin/Package.resolved') }}
```
Comment (KaterinaWire) [inline .github/workflows/_reusable_app_release.yml:174]: the key here is `macOS-swiftpm-sourcery-`

**Comment by KaterinaWire** [comment]
@netbe and @caldrian Please review this PR

File: fastlane/Fastfile [hunk 6]
```diff
@@ -350,6 +350,7 @@ platform :ios do
             xcargs += enable_datadog_args if build.enable_datadog
             xcargs += ["COUNTLY_APP_KEY=#{build.countly_api_key}"] if build.countly_enabled
 
+           sh "echo $HAS_PACKAGES_CACHE_HIT"
```
Comment (netbe) [inline fastlane/Fastfile]:
```suggestion
            sh "echo $HAS_PACKAGES_CACHE_HIT"
```

Comment (netbe) [reply fastlane/Fastfile]: is it aligned?

File: wire-ios-system/WireSystem.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (netbe) [inline wire-ios-system/WireSystem.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:1]: why is this file deleted? I wonder if we need the package.resolved of each package.

**Review by netbe — APPROVED** [review]
left two minor comments, but overall this looks good! thanks

File: wire-ios-system/WireSystem.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (caldrian) [reply wire-ios-system/WireSystem.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:1]: Since we always only open the workspace (`wire-ios-mono.xcworkspace`) we only need the Package.resolved there.

**Review by caldrian — APPROVED** [review]
